3-java-servlet-web-app/1-Authentication/sign-in-ciam/README.md

```diff
@@ -0,0 +1,272 @@
+
+## We'd love your feedback!
+
+Were we successful in addressing your learning objective? Consider taking a moment to [share your experience with us](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR73pcsbpbxNJuZCMKN0lURpURDQwVUxQWENUMlpLUlA0QzdJNVE3TUJRSyQlQCN0PWcu).
+
+## About the code
+
+This sample shows how to use **MSAL for Java (MSAL4J)** to sign in users into your Microsoft Entra tenant. If you'd like to use MSAL4J in your own applications, must add it to your projects using Maven.
+
+If you want to replicate this sample's behavior, you may choose to copy the `pom.xml` file, and the contents of the `helpers` and `authservlets` packages in the `src/main/java/com/microsoft/azuresamples/msal4j` package. You'll also need the [authentication.properties file](src/main/resources/authentication.properties). These classes and files contain generic code that can be used in a wide array of applications. The rest of the sample may be copied as well, but the other classes and files are built specifically to address this sample's objective.
+
+A **ConfidentialClientApplication** instance is created in the [AuthHelper.java](src/main/java/com/microsoft/azuresamples/authentication/AuthHelper.java) class. This object helps craft the Microsoft Entra authorization URL and also helps exchange the authentication token for an access token.
+
+```Java
+// getConfidentialClientInstance method
+IClientSecret secret = ClientCredentialFactory.createFromSecret(SECRET);
+confClientInstance = ConfidentialClientApplication
+                    .builder(CLIENT_ID, secret)
+                    .authority(AUTHORITY)
+                    .build();
+```
+
+The following parameters need to be provided upon instantiation:
+
+- The **Client ID** of the app
+- The **Client Secret**, which is a requirement for Confidential Client Applications
+- The **Microsoft Entra authority**, which includes your Microsoft Entra tenant ID.
+
+In this sample, these values are read from the [authentication.properties](src/main/resources/authentication.properties) file using a properties reader in the class [Config.java](src/main/java/com/microsoft/azuresamples/authentication/Config.java).
+
+### Step-by-step walkthrough
+
+1. The first step of the sign-in process is to send a request to the `/authorize` endpoint on for our Microsoft Entra tenant. Our MSAL4J `ConfidentialClientApplication` instance is leveraged to construct an authorization request URL. Our app redirects the browser to this URL, which is where the user will sign in.
+
+    ```Java
+    final ConfidentialClientApplication client = getConfidentialClientInstance();
+    AuthorizationRequestUrlParameters parameters = AuthorizationRequestUrlParameters.builder(Config.REDIRECT_URI, Collections.singleton(Config.SCOPES))
+            .responseMode(ResponseMode.QUERY).prompt(Prompt.SELECT_ACCOUNT).state(state).nonce(nonce).build();
```

Comment:
Is "SELECT_ACCOUNT" prompt needed here? Does the token issuer's default value not provide a good experience?

Reply:
This was copied from the basic sign-in sample, and after trying with and without it definitely doesn't seem needed. Removed in the latest commit, and I'll be going through the other samples too at some point.